spatialdata/_core/_spatial_query.py

```diff
@@ -57,154 +187,365 @@ def __post_init__(self) -> None:
+        min_coordinate=min_coordinate,
+        max_coordinate=max_coordinate,
+    )
     from spatialdata._core._spatialdata_ops import (
         get_transformation,
         set_transformation,
     )
 
+    # get the transformation from the element's intrinsic coordinate system to the query coordinate space
+    transform_to_query_space = get_transformation(image, to_coordinate_system=target_coordinate_system)
+    assert isinstance(transform_to_query_space, BaseTransformation)
+    m = _get_affine_for_element(image, transform_to_query_space)
+    input_axes_without_c = tuple([ax for ax in m.input_axes if ax != "c"])
+    output_axes_without_c = tuple([ax for ax in m.output_axes if ax != "c"])
+    m_without_c = m.to_affine_matrix(input_axes=input_axes_without_c, output_axes=output_axes_without_c)
+    m_without_c_linear = m_without_c[:-1, :-1]
+
+    transform_dimension = np.linalg.matrix_rank(m_without_c_linear)
+    transform_coordinate_length = len(output_axes_without_c)
+    data_dim = len(input_axes_without_c)
+
+    assert data_dim in [2, 3]
+    assert transform_dimension in [2, 3]
+    assert transform_coordinate_length in [2, 3]
+    assert not (data_dim == 2 and transform_dimension == 3)
+    assert not (transform_dimension == 3 and transform_coordinate_length == 2)
+    # see explanation in https://github.com/scverse/spatialdata/pull/151
+    if data_dim == 2 and transform_dimension == 2 and transform_coordinate_length == 2:
+        case = 1
+    elif data_dim == 2 and transform_dimension == 2 and transform_coordinate_length == 3:
+        case = 2
+    elif data_dim == 3 and transform_dimension == 2 and transform_coordinate_length == 2:
+        case = 3
+    elif data_dim == 3 and transform_dimension == 2 and transform_coordinate_length == 3:
+        case = 4
+    elif data_dim == 3 and transform_dimension == 3 and transform_coordinate_length == 3:
+        case = 5
+    else:
+        raise RuntimeError("This should not happen")
+
+    if set(axes) != set(output_axes_without_c):
+        if set(axes).issubset(output_axes_without_c):
+            logger.warning(
+                f"The element has axes {output_axes_without_c}, but the query has axes {axes}. Excluding the element "
+                f"from the query result. In the future we can add support for this case. If you are interested, "
+                f"please open a GitHub issue."
+            )
+            return None
+        else:
+            raise ValueError(
+                f"Invalid case. The bounding box axes are {axes}, the spatial axes in {target_coordinate_system} are {output_axes_without_c}"
+            )
+    spatial_transform = Affine(m_without_c, input_axes=input_axes_without_c, output_axes=output_axes_without_c)
+    spatial_transform_bb_axes = Affine(
+        spatial_transform.to_affine_matrix(input_axes=input_axes_without_c, output_axes=axes),
+        input_axes=input_axes_without_c,
+        output_axes=axes,
+    )
+    if case in [1, 2, 5]:
```

Comment:
only computing `bounding_box_corners` is different from case 1+5 and case 2. The second part of the code is shared. But now I will simplify since we don't support case 3 and 4 anymore (as discussed via chat).